curiosrc/ffi/sdr_funcs.go

```diff
@@ -625,3 +587,87 @@ func (sb *SealCalls) sectorStorageType(ctx context.Context, sector storiface.Sec
 
 	return true, storiface.PathStorage, nil
 }
+
+// PreFetch fetches the sector file to local storage before SDR and TreeRC Tasks
+func (sb *SealCalls) PreFetch(ctx context.Context, sector storiface.SectorRef, task *harmonytask.TaskID) error {
+	_, _, releaseSector, err := sb.sectors.AcquireSector(ctx, task, sector, storiface.FTCache, storiface.FTSealed, storiface.PathSealing)
+	if err != nil {
+		return xerrors.Errorf("acquiring sector paths: %w", err)
+	}
+	defer releaseSector()
+
+	return nil
+}
+
+func (sb *SealCalls) TreeD(ctx context.Context, task *harmonytask.TaskID, sector storiface.SectorRef, unsealed cid.Cid, size abi.PaddedPieceSize, data io.Reader, unpaddedData bool) error {
+	fspaths, pathIDs, releaseSector, err := sb.sectors.AcquireSector(ctx, task, sector, storiface.FTCache, storiface.FTSealed, storiface.PathSealing)
+	if err != nil {
+		return xerrors.Errorf("acquiring sector paths: %w", err)
+	}
+	defer releaseSector()
```

Comment:
Due to how reservations work you can only call `sb.sectors.AcquireSector` once per task.

The first acquire will acquire from the (task_storage) reservation, then the call to `releaseSector` will release the reservation; the second acquire may then fail if another task happens to be scheduled in between.